lua/typescript-tools/config.lua

```diff
@@ -178,6 +180,13 @@ function M.load_settings(settings)
   if not M.code_lens_mode[settings.code_lens] then
     __store.code_lens = M.code_lens_mode.off
   end
+
+  if not settings.jsx_close_tag then
+    __store.jsx_close_tag = {
+      enable = false,
+      filetypes = { "javascriptreact", "typescriptreact" },
+    }
+  end
```

Comment:
This allow to do partial application to this setting.
When someone set only enable to true this automatically set default filetypes for it.

```suggestion
  local default_jsx_filetypes = { "javascriptreact", "typescriptreact" }
  
  if not settings.jsx_close_tag then
    __store.jsx_close_tag = {
      enable = false,
      filetypes = default_jsx_filetypes,
    }
  elseif settings.jsx_close_tag and not settings.jsx_close_tag.filetypes then
      __store.jsx_close_tag.filetypes = default_jsx_filetypes
  end
```

Reply:
fixed in https://github.com/pmizio/typescript-tools.nvim/pull/185/commits/a984e5fe5e514597ed3db4711dbcf01e22215ca9